### PR TITLE
Enable creation of ResourceLoaderUrlFormatter for external contexts

### PR DESCRIPTION
**Pull request by tbragaf**

### CODE REVIEW

File: ViewGenerator/contentFiles/global.d.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ declare module "ResourceLoader" {
     export type ResourceLoaderUrlFormatter = (resourceKey: string, ...params: string[]) => string;
 
     export const ResourceLoader: React.Context<ResourceLoaderUrlFormatter>;
+    export function createResourceLoaderUrlFormatter(customResourceBaseUrl: string, viewName: string): ResourceLoaderUrlFormatter;
```
Comment (joaompneves) [inline ViewGenerator/contentFiles/global.d.ts]: customResourceBaseUrl ... shouldn't be exposed

Comment (joaompneves) [reply ViewGenerator/contentFiles/global.d.ts]: Actually expose a function that returns the url

```
export function formatUrl(viewName: string, resourceKey: string, ...params: string[]): string{
 const urlTail = [resourceKey].concat(params).map(p => encodeURIComponent(p)).join("&");
 return `${customResourceBaseUrl}/${viewName}/?${urlTail}`;
}
```

Comment (joaompneves) [reply ViewGenerator/contentFiles/global.d.ts]: Store the customResourceBaseUrl in a local variable that should be set on the Loader (using the url params already grabbed)

File: ReactViewResources/Loader/Loader.View.tsx [hunk 2]
```diff
@@ -4,8 +4,7 @@ import { PluginsContext, PluginsContextHolder } from "./PluginsContext";
 import { ViewContext } from "./ViewContext";
 import { ViewMetadata } from "./ViewMetadata";
 import { ViewPortalsCollection, ViewLifecycleEventHandler } from "./ViewPortalsCollection";
-import { ResourceLoader } from "./ResourceLoader";
-import { ResourceLoaderUrlFormatter } from "ResourceLoader";
+import { ResourceLoader, createResourceLoaderUrlFormatter } from "./ResourceLoader";
 
 export function createView(
```
Comment (joaompneves) [inline ReactViewResources/Loader/Loader.View.tsx]: remove the customResourceBaseUrl param

**Comment by joaompneves** [comment]
don't forget to raise react view version

File: ReactViewResources/Loader/ResourceLoader.ts [hunk 3]
```diff
@@ -1,6 +1,16 @@
 ﻿import * as React from "react";
 import { ResourceLoaderUrlFormatter } from "ResourceLoader";
 
+let customResourceBaseUrl = "";
+export function setCustomResourceBaseUrl(url: string): void {
```
Comment (joaompneves) [inline ReactViewResources/Loader/ResourceLoader.ts]: add line

File: ViewGenerator/contentFiles/global.d.ts (hunk 1 above)
Comment (tbragaf) [reply ViewGenerator/contentFiles/global.d.ts]: Done

File: ReactViewResources/Loader/ResourceLoader.ts (hunk 3 above)
Comment (tbragaf) [reply ReactViewResources/Loader/ResourceLoader.ts]: Done

**Pull request merged**